src/schemathesis/loaders.py

```diff
@@ -54,13 +55,18 @@ def from_uri(
     endpoint: Optional[Filter] = None,
     tag: Optional[Filter] = None,
     operation_id: Optional[Filter] = None,
+    port: Optional[int] = None,
     *,
     app: Any = None,
     validate_schema: bool = True,
     **kwargs: Any,
 ) -> BaseOpenAPISchema:
     """Load a remote resource and parse to schema instance."""
     kwargs.setdefault("headers", {}).setdefault("User-Agent", USER_AGENT)
+    if not base_url and port:
+        o = urlparse(uri)
```

Comment:
Let's use "parsed" as the variable name

Reply:
This will be removed in favor of using `yarl`.